File: app/blog/internal/service/post.go
package service

import (
	"context"
	"github.com/go-kratos/kratos/v2/log"
	"google.golang.org/protobuf/types/known/emptypb"
	v1 "kratos-blog/api/blog/v1"
	"kratos-blog/app/blog/internal/biz"
	"kratos-blog/third_party/pagination"
)

type PostService struct {
	v1.UnimplementedPostServiceServer

	uc  *biz.PostUseCase
	log *log.Helper
}

func NewPostService(logger log.Logger, uc *biz.PostUseCase) *PostService {
	l := log.NewHelper(log.With(logger, "module", "service/post"))
	return &PostService{
		log: l,
		uc:  uc,
	}
}

// ListPost 获取帖子列表
func (s *PostService) ListPost(ctx context.Context, req *pagination.PagingRequest) (*v1.ListPostResponse, error) {
	return s.uc.List(ctx, req)
}

// GetPost 获取帖子数据
func (s *PostService) GetPost(ctx context.Context, req *v1.GetPostRequest) (*v1.Post, error) {
	return s.uc.Get(ctx, req)
}

// CreatePost 创建帖子
func (s *PostService) CreatePost(ctx context.Context, req *v1.CreatePostRequest) (*v1.Post, error) {
	return s.uc.Create(ctx, req)
}

// UpdatePost 更新帖子
func (s *PostService) UpdatePost(ctx context.Context, req *v1.UpdatePostRequest) (*v1.Post, error) {
	return s.uc.Update(ctx, req)
}

// DeletePost 删除帖子
func (s *PostService) DeletePost(ctx context.Context, req *v1.DeletePostRequest) (*emptypb.Empty, error) {
	_, err := s.uc.Delete(ctx, req)
	if err != nil {
		return nil, err
	}
	return &emptypb.Empty{}, nil
}

File: blog-frontend/blog-admin/src/api/blog/post.ts
import { defHttp } from '/@/utils/http/axios';

const route = {
  Posts: '/posts',
  PostWithId: (id) => `/posts/${id}`,
};

// 获取列表
export const ListPost: ListPost = (params) => {
  return defHttp.get<ListPostResponse>({
    url: route.Posts,
    params,
  });
};

// 获取
export const GetPost: GetPost = (params) => {
  return defHttp.get<Post>({
    url: route.PostWithId(params.id),
  });
};

// 创建
export const CreatePost: CreatePost = (params) => {
  return defHttp.post<Post>(
    { url: route.Posts, params },
    {
      errorMessageMode: 'none',
    },
  );
};

// 更新
export const UpdatePost: UpdatePost = (params) => {
  return defHttp.put<Post>({
    url: route.PostWithId(params.id),
    data: params.post,
  });
};

// 删除
export const DeletePost: DeletePost = (params) => {
  return defHttp.delete({
    url: route.PostWithId(params.id),
  });
};
